dispatch_test.go

```diff
@@ -62,10 +62,15 @@ func TestDispatchCanceledBeforeResult(t *testing.T) {
 
 	cancel()
 
+	// Should not hang.
+	sendResponse(c, nil)
+
 	err := <-done
-	require.EqualError(t, err, context.Canceled.Error())
+	require.True(t, err == nil || err == context.Canceled,
```

Comment:
Since now the channel send isn't synchronous, the select in Call can pick either one of the two options regardless of which one is done first.